onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidatePassword.kt

```diff
@@ -0,0 +1,28 @@
+package com.alexafit.onboardingauthdomain.useCase
+
+class ValidatePassword {
+
+    operator fun invoke(password: String): Pair<String, Boolean> {
+        return when (password.isBlank()) {
+            true -> Pair(password.trim(), false)
+            false -> {
+                val trimmedPassword = password.trimEnd()
+                trimmedPassword.isPasswordValid()
+            }
+        }
+    }
+
+    private fun String.isPasswordValid(): Pair<String, Boolean> {
+        return if (this.length >= 9) {
+            val hasLowercaseLetter = this.any { it.isLowerCase() }
+            val hasUppercaseLetter = this.any { it.isUpperCase() }
+            val hasDigit = this.any { it.isDigit() }
+            val hasNoSpecialCharacter = this.all { it.isLetterOrDigit() }
```

Comment:
Why isn't a password allowed to have a special character? It's not a requirement to have one, but also not illegal